diff --git libbb/lineedit.c libbb/lineedit.c
index fbabc6c..c3ca551 100644
--- libbb/lineedit.c
+++ libbb/lineedit.c
@@ -72,7 +72,8 @@
 # define CHAR_T wchar_t
 static bool BB_isspace(CHAR_T c)
 {
-	return ((unsigned)c < 256 && isspace(c));
+	/* Total lie, but better matches the behavior of other shells and vi */
+	return ((unsigned)c < 256 && isspace(c)) || c == '/' || c == '-';
 }
 # if ENABLE_FEATURE_EDITING_VI
 static bool BB_isalnum_or_underscore(CHAR_T c)
@@ -95,7 +96,7 @@ static bool BB_ispunct(CHAR_T c)
 #else
 # define BB_NUL '\0'
 # define CHAR_T char
-# define BB_isspace(c) isspace(c)
+# define BB_isspace(c) isspace(c) || c == '/' || c == '-'
 # if ENABLE_FEATURE_EDITING_VI
 static bool BB_isalnum_or_underscore(CHAR_T c)
 {
