diff --git src/nnn.h src/nnn.h
index adebd69..e2201ab 100644
--- src/nnn.h
+++ src/nnn.h
@@ -120,8 +120,10 @@ static struct key bindings[] = {
 	{ KEY_UP,         SEL_PREV },
 	/* Page down */
 	{ KEY_NPAGE,      SEL_PGDN },
+	{ CONTROL('F'),   SEL_PGDN },
 	/* Page up */
 	{ KEY_PPAGE,      SEL_PGUP },
+	{ CONTROL('B'),   SEL_PGUP },
 	/* Ctrl+D */
 	{ CONTROL('D'),   SEL_CTRL_D },
 	/* Ctrl+U */
@@ -168,7 +170,7 @@ static struct key bindings[] = {
 	{ 'd',            SEL_DETAIL },
 	/* File details */
 	{ 'f',            SEL_STATS },
-	{ CONTROL('F'),   SEL_STATS },
+	{ 'F',            SEL_STATS },
 	/* Toggle executable status */
 	{ '*',            SEL_CHMODX },
 	/* Create archive */
