+++ src/nnn.h
@@ -133,8 +133,10 @@
 	{ KEY_UP,         SEL_PREV },
 	/* Page down */
 	{ KEY_NPAGE,      SEL_PGDN },
+	{ CONTROL('F'),   SEL_PGDN },
 	/* Page up */
 	{ KEY_PPAGE,      SEL_PGUP },
+	{ CONTROL('B'),   SEL_PGUP },
 	/* Ctrl+D */
 	{ CONTROL('D'),   SEL_CTRL_D },
 	/* Ctrl+U */
@@ -189,7 +191,7 @@
 	/* List archive */
 	{ 'F',            SEL_ARCHIVELS },
 	/* Extract archive */
-	{ CONTROL('F'),   SEL_EXTRACT },
+	{ 'x',            SEL_EXTRACT },
 	/* Toggle sort by size */
 	{ 'z',            SEL_FSIZE },
 	/* Sort by apparent size including dir contents */
@@ -248,7 +250,6 @@
 	{ '!',            SEL_SHELL },
 	{ CONTROL(']'),   SEL_SHELL },
 	/* Plugin key */
-	{ 'x',            SEL_PLUGKEY },
 	{ ':',            SEL_PLUGKEY },
 	{ ';',            SEL_PLUGKEY },
 	/* Run a plugin */
